modules/skybrowser/skybrowsermodule.cpp

```diff
@@ -0,0 +1,692 @@
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/skybrowser/skybrowsermodule.h>
+
+#include <modules/skybrowser/include/screenspaceskybrowser.h>
+#include <modules/skybrowser/include/targetbrowserpair.h>
+#include <modules/skybrowser/include/screenspaceskytarget.h>
+#include <modules/skybrowser/include/wwtdatahandler.h>
+#include <modules/base/rendering/screenspaceimagelocal.h>
+#include <openspace/scene/scene.h>
+#include <openspace/scene/scenegraphnode.h>
+#include <openspace/engine/globalscallbacks.h>
+#include <openspace/engine/windowdelegate.h>
+#include <openspace/navigation/navigationhandler.h>
+#include <openspace/camera/camera.h>
+#include <openspace/util/factorymanager.h>
+
+#include "skybrowsermodule_lua.inl"
+
+namespace {
+    constexpr const openspace::properties::Property::PropertyInfo AllowInteractionInfo =
+    {
+        "AllowMouseInteraction",
+        "Allow Mouse Interaction",
+        "Toggles if it is possible to interact with the sky browser and sky targets with "
+        "the mouse or not."
+    };
+    
+    constexpr const openspace::properties::Property::PropertyInfo AllowRotationInfo =
+    {
+        "AllowCameraRotation",
+        "Allow Camera Rotation",
+        "Toggles if the camera should rotate to look at the sky target if it is going "
+        "outside of the current field of view."
+    };
+
+    struct [[codegen::Dictionary(SkyBrowserModule)]] Parameters {
+        // [[codegen::verbatim(AllowInteractionInfo.description)]]
+        std::optional<bool> allowMouseInteraction;
+
+        // [[codegen::verbatim(AllowRotationInfo.description)]]
+        std::optional<bool> allowCameraRotation;
+    };
+    
+    #include "skybrowsermodule_codegen.cpp"
+} // namespace
+
+namespace openspace {
+    scripting::LuaLibrary SkyBrowserModule::luaLibrary() const {
+        scripting::LuaLibrary res;
+        res.name = "skybrowser";
+        res.functions = {
+            {
+                "getListOfImages",
+                &skybrowser::luascriptfunctions::getListOfImages,
+                "",
+                "Returns a list of all the loaded AAS WorldWide Telescope images that "
+                "has been loaded."
+            }, 
+            {
+                "setHoverCircle",
+                &skybrowser::luascriptfunctions::setHoverCircle,
+                "string",
+                "Takes an identifier to a screen space renderable and adds it to the "
+                "module."
+            },
+            {
+                "moveCircleToHoverImage",
```

Comment:
Again, should it be called hover "circle", or something more generic, like "marker"?